File: app/src/main/java/com/example/mtgcreaturesearch/Model/Creature.java
package com.example.mtgcreaturesearch.Model;

import org.jetbrains.annotations.Nullable;

public class Creature {

    private String name;
    private boolean favorited;
    @Nullable

    public Creature(String creatureName, boolean favorite){
        name = creatureName;
        favorited = favorite;
    }

    public void setFavorited(boolean favorite) {
        favorited = favorite;
    }

    public String getName() {
        return name;
    }

    public boolean getFavorited() {
        return favorited;
    }
}

File: app/src/main/java/com/example/mtgcreaturesearch/model/Creature.kt
package com.example.mtgcreaturesearch.model

import androidx.compose.runtime.mutableStateOf

class Creature constructor(creatureName: String, favorite: Boolean){

    private val name: String = creatureName
    private var _favorited = mutableStateOf(favorite)
    private var favorited: Boolean = _favorited.value

    fun setFavorited(favorite: Boolean) {
        favorited = favorite
    }

    fun getName(): String {
        return name
    }

    fun getFavorited(): Boolean {
        return favorited
    }

}